blockchain/validate.go

```diff
@@ -46,6 +46,14 @@ const (
 	// MaxCoinbaseScriptLen is the maximum length a coinbase script can be.
 	MaxCoinbaseScriptLen = 100
 
+	// MinTSpendScriptLen is the minimum length a treasurybase script can
+	// be.
+	MinTSpendScriptLen = 45
+
+	// MaxTPendScriptLen is the maximum length a treasurybase script can
+	// be.
+	MaxTSpendScriptLen = 125
```

Comment:
Those were the values @davecgh and I landed on. We went back and forth on being exact vs allowing some margin and possible size differences. @davecgh would you like to chime in here?

Reply:
We landed on those for DER signatures.   For Schnorr, they are indeed different.  In fact, these can go away and we can just limit the script to exactly 100 bytes now.

`DATA_64 <64-byte Schnorr signature> DATA_33 <33-byte compressed pubkey> TSPEND`

1+64+1+33+1 = 100

Notice that there is no hash type on the tspend because we explicitly want to ensure that it is always `SigHashAll`, so that is intentionally hard coded as part of the consensus rules.

For reference, the minimum of 45 came from the fact that the smallest theoretical possible DER signature would be 8 bytes and the script previously included the signature hash type for an additional byte.  So it would've been:

`DATA_9 <8-byte DER signature + 1 byte signature hash type> DATA_33 <33-byte compressed pubkey> TSPEND`

1+9+1+33+1 = 45

However, as noted above, we decided to explicitly force `SigHashAll` as part of the rules and hence there is no reason to include the extra byte.

Reply:
I added a constant now that we use Schnorr.